pkg/runexec/runexec.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runexec
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/dapr/cli/pkg/standalone"
+)
+
+type CmdProcess struct {
+	Command      *exec.Cmd
+	CommandErr   error
+	OutputWriter io.Writer
+	ErrorWriter  io.Writer
+}
+
+type RunExec struct {
+	DaprCMD        *CmdProcess
+	AppCMD         *CmdProcess
+	AppID          string
+	DaprHTTPPort   int
+	DaprGRPCPort   int
+	DaprMetricPort int
+}
+
+// RunOutput represents the run execution.
+type RunOutput struct {
```

Comment:
This will later be removed, once the normal `dapr run` also migrates to using the `RunExec` struct.